test/unit/lapack/gpu/test_larft.cpp

```diff
@@ -0,0 +1,96 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2024, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include <tuple>
+#include <vector>
+
+#include <whip.hpp>
+
+#include <dlaf/blas/enum_output.h>
+#include <dlaf/common/single_threaded_blas.h>
+#include <dlaf/gpu/blas/error.h>
+#include <dlaf/lapack/gpu/larft.h>
+#include <dlaf/util_matrix.h>
+
+#include <gtest/gtest.h>
+
+#include <dlaf_test/matrix/util_tile.h>
+#include <dlaf_test/util_types.h>
+
+using namespace dlaf;
+using namespace dlaf::test;
+using dlaf::matrix::test::createTile;
+using dlaf::matrix::util::internal::getter_random;
+
+template <class T>
+struct LarftTestGPU : public ::testing::Test {};
+
+TYPED_TEST_SUITE(LarftTestGPU, MatrixElementTypes);
+
+std::vector<std::tuple<SizeType, SizeType, SizeType, SizeType>> configs{
+    // n, k, mb, nb, v_start
+    {0, 0, 1, 1},       {0, 4, 4, 4},         {4, 0, 5, 5},                       // Empty
+    {6, 6, 6, 6},       {5, 3, 5, 3},         {7, 3, 10, 13},      {3, 7, 8, 8},  // Very Small
+    {127, 35, 128, 36}, {128, 128, 128, 128}, {96, 127, 256, 256},                // A bit larger
+};
+
+TYPED_TEST(LarftTestGPU, CorrectnessLocal) {
+  // Note:
+  // There are two main differences of GPU larft compared to LAPACK implementation:
+  // 1) the first element of the reflectors has to be set to 1. (implicit considered as one in LAPACK)
+  // 2) The lower part of the T factor is 0. (untouched in LAPACK)
```

Comment:
Is this an assumption that could be commented in `larft.cu` as well/instead?

Reply:
File sneaked in from #514... removed